### ISSUE TITLE
[BUG][Windows] Wierd error for some path arguments with a space inside

### ISSUE THREAD

**Issue by znjameswu** [tekartik/process_run.dart#20]
## Demo
```
import 'package:process_run/cmd_run.dart';
import 'package:process_run/process_run.dart';

void main(List<String> arguments) async{
  final version = 'master';
  final kFlutterRepo = 'https://github.com/flutter/flutter.git';
  final versionDirectoryPath = r'C:\Users\FIRSTNAME LASTNAME\fvm\versions\master';
  final args = [
    'clone',
    '-b',
    version,
    kFlutterRepo,
    versionDirectoryPath
  ];
  final process = await run(
    'git',
    args,
    verbose: true,
  );
}
```
Output:
```
$ git clone -b master https://github.com/flutter/flutter.git "C:\Users\FIRSTNAME LASTNAME\fvm\versions\master"
'C:\Program' is not recognized as an internal or external command,
operable program or batch file.
```

### Special points:
1. The logged git command is perfectly legal
2. Whatever path you give it, the error will always start with 'C:\Program'
3. Not every command will crash on a path with a space. Currently I only found `git clone`. For example, `run('dart', [SOME_PATH])` works perfectly on my machine

### Environment
`Dart SDK version: 2.9.1 (stable) (Wed Aug 12 12:37:19 2020 +0200) on "windows_x64"`
process_run version: "0.10.12+1"
git version 2.24.0.windows.2
### Related
https://github.com/leoafarias/fvm/issues/103

**Comment by alextekartik** [comment]
Thanks for report that I was able to reproduce on windows (after some puzzling effort, we need a space in the executable and in the arguments). The issue was that runInShell was force to true which was causing this weird crash.
Fixed in 0.10.12+2 just published although you could alternatively set runInShell to false to workaround in your current version.

**Comment by znjameswu** [comment]
I can confirm this issuee is fixed. Thank you!
